### PR TITLE
OCPBUGS-33082: Add new use cases for networking obfuscation

### PR DESCRIPTION
**Pull request by ncaak**
This PR implements a new fix for previous PR #936 

## Categories

- [X] Bugfix
- [ ] Data Enhancement
- [ ] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `path/to/sample_data.json`

## Documentation

- `path/to/documentation.md`

## Unit Tests

- `path/to/file_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Changelog

No

## Breaking Changes

Yes

## References

https://issues.redhat.com/browse/OCPBUGS-33082

### CODE REVIEW

**Comment by JoaoFula** [comment]
techpreview tests are passing. Hiccup is on cluster de-provisioning. Feel free to override the tests if necessary.

File: pkg/anonymization/anonymizer.go [hunk 1]
```diff
@@ -310,15 +307,18 @@ func NewAnonymizerFromConfig(
 	configurator configobserver.Interface,
 	dataPolicy v1alpha1.DataPolicy,
 ) (*Anonymizer, error) {
+	sensitiveVals := make(map[string]string)
 	kubeClient, err := kubernetes.NewForConfig(protoKubeConfig)
 	if err != nil {
 		return nil, err
 	}
+	sensitiveVals[extractDomain(protoKubeConfig.Host)] = ClusterHostPlaceholder
```
Comment (tremes) [inline pkg/anonymization/anonymizer.go:315]: Interesting idea using the kubeconfig data! :+1:

**Comment by tremes** [comment]
I didn't try that, but 
/lgtm

**Pull request merged**